sdk/cosmos/azure-cosmos-benchmark/src/main/java/com/azure/cosmos/benchmark/AsyncWriteBenchmark.java

```diff
@@ -60,13 +61,13 @@ protected void hookOnError(Throwable throwable) {
     @Override
     protected void performWorkload(BaseSubscriber<CosmosAsyncItemResponse> baseSubscriber, long i) throws InterruptedException {
         String partitionKey = uuid + i;
-        Mono<CosmosAsyncItemResponse> obs;
+        Mono<CosmosAsyncItemResponse<PojoizedJson>> obs;
         if (configuration.isDisablePassingPartitionKeyAsOptionOnWrite()) {
             // require parsing partition key from the doc
             obs = cosmosAsyncContainer.createItem(generateDocument(partitionKey, dataFieldValue));
         } else {
             // more optimized for write as partition ke is already passed as config
-            obs = cosmosAsyncContainer.createItem(generateDocument(partitionKey, dataFieldValue), new CosmosItemRequestOptions(partitionKey));
+            obs = cosmosAsyncContainer.createItem(generateDocument(partitionKey, dataFieldValue), new CosmosItemRequestOptions());
```

Comment:
we should pass partition key to CosmosItemRequestOption here.

The intention of this benchmark is to measure perf when pk is passed as request options.

Reply:
Added new API accept partition key and changed this call accordingly.